src/udp.rs

```diff
@@ -39,103 +114,293 @@ impl UdpSocket {
         bus: &mut SpiBus,
         remote: SocketAddrV4,
     ) -> Result<(), UdpSocketError<SpiBus::Error>> {
-        self.socket.set_destination_ip(bus, *remote.ip())?;
-        self.socket.set_destination_port(bus, remote.port())?;
+        // We set this variable to true when:
+        /// - We have the same Socket address already set
+        /// - We don't have any previous destination set
+        let is_same = self
+            .destination
+            .filter(|set_destination| set_destination == &remote)
+            .is_some();
+
+        // either no previous destination is set or a new one
+        if !is_same {
+            self.socket.set_destination_ip(bus, *remote.ip())?;
+            self.socket.set_destination_port(bus, remote.port())?;
+
+            // set the new destination after we've successfully set them on chip
+            self.destination = Some(remote);
+        }
+
+        Ok(())
+    }
+
+    fn socket_send_all<SpiBus: Bus>(
+        &self,
+        bus: &mut SpiBus,
+        send_buffer: &[u8],
+    ) -> NbResult<(), UdpSocketError<SpiBus::Error>> {
+        match Status::try_from(self.socket.get_status(bus)?) {
+            Ok(Status::Udp) => {}
+            Ok(status) => return Err(NbError::Other(UdpSocketError::SocketNotOpen)),
+            Err(err) => return Err(NbError::Other(UdpSocketError::UnrecognisedStatus)),
+        }
+
+        if self.destination.is_none() {
+            return Err(NbError::Other(UdpSocketError::DestinationNotSet));
+        }
+
+        let mut free_size = self.socket.get_tx_free_size(bus)?;
+
+        // Ensure write is currently possible.
+        // This should never be `0`
+        if free_size == 0 {
+            // If this happens, then `Send` was not called in previous operations.
+            // FIXME: add a way to either:
```

Comment:
Can you spawn an issue for this if it's still relevant?

Reply:
We'll leave this in for now. If someone hits an issue around it, they can spawn an issue in the repo